docs/test-specs/sparse_merkle_tree_tests.md

```diff
@@ -0,0 +1,396 @@
+
+The tests described in this document are designed to test features common to most Sparse Merkle Tree implementations. Test specifications are agnostic of the implementation details or language, and therefore take a black-box testing approach. A test specification may provide an example of what a compliant test may look like in the form of pseudocode.
+
+A test specification follows the format:
+- Test name
+- Test description
+- Test inputs
+- Test outputs
+- Example pseudocode
+
+For a concrete test to comply with its corresponding test specification, the System Under Test (SUT) must take in the prescribed inputs. When the SUT produces the prescribed outputs, the test passes. When the SUT produces any result or error that is not prescribed by the speciifcation, the test fails. For a library to comply with the complete specification described herein, it must implement all test specifications, and each test must pass.
+
+All test specifications assume that the Merkle Tree implementation under test uses the SHA-2-256 hashing algorithm as defined in [FIPS PUB 180-4](https://doi.org/10.6028/NIST.FIPS.180-4) to produce its outputs.
+
+## Root Signature Tests
+
+1. [Test Empty Root](#test-empty-root)
+2. [Test Root Update 1](#test-root-update-1)
+3. [Test Root Update 2](#test-root-update-2)
+4. [Test Root Update 3](#test-root-update-3)
+5. [Test Root Update 5](#test-root-update-5)
+6. [Test Root Update 10](#test-root-update-10)
+7. [Test Root Update 100](#test-root-update-100)
+8. [Test Root Update Empty With Null Data](#test-root-update-empty-with-null-data)
+9. [Test Root Update With Null Data Performs Delete](#test-root-update-with-null-data-performs-delete)
+10. [Test Root Update 1 Delete 1](#test-root-update-1-delete-1)
+11. [Test Root Update 2 Delete 1](#test-root-update-2-delete-1)
+12. [Test Root Update 10 Delete 5](#test-root-update-10-delete-5)
+13. [Test Root Interleaved Update Delete](#test-root-interleaved-update-delete)
+---
+
+### Test Empty Root
+
+**Description**:
+
+Tests the default root given no update or delete operations.
+
+**Inputs**:
+
+_No inputs_
+
+**Outputs**:
+
+- The expected root signature: `0xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
```

Comment:
Should we mark this part of the spec as WIP until the corresponding issue in celestia (https://github.com/celestiaorg/smt/issues/67) is resolved?

Reply:
Yes, I think that's a good interim solution. I will add that affected tests are WIP and reference the issue in this document. Once we've determined the correct root, I will update those sections, and bump the document version.